src/GovUk.Frontend.AspNetCore/TagHelpers/CharacterCountTagHelper.cs

```diff
@@ -254,8 +252,32 @@ private protected override IHtmlContent GenerateFormGroupContent(
         var textAreaTagBuilder = GenerateTextArea(haveError);
         contentBuilder.AppendHtml(textAreaTagBuilder);
 
+        var countRemainingMessage = GenerateCountRemainingMessage();
+        contentBuilder.AppendHtml(countRemainingMessage);
+
         return contentBuilder;
 
+
+        IHtmlContent GenerateCountRemainingMessage()
+        {
+            // TODO / FIXME: temporary hardcoded, need to pass through from somewhere? Unable to access context from here?
+            string textAreaDescriptionText = null;
+            var hasNoLimit = MaxLength is null && MaxWords is null; // TODO: This logic is reused in many places -- centralise it?
```

Comment:
Assuming the `*Context` could be made available, is it appropriate to centralise the `hasNoLimit` logic to it? Somewhere else, perhaps?